implementation/egads/src/main/java/net/opentsdb/query/anomaly/egads/olympicscoring/OlympicScoringBaseline.java

```diff
@@ -0,0 +1,251 @@
+import net.opentsdb.data.types.numeric.NumericArrayType;
+import net.opentsdb.data.types.numeric.NumericSummaryType;
+import net.opentsdb.data.types.numeric.NumericType;
+import net.opentsdb.query.QueryResult;
+import net.opentsdb.query.anomaly.egads.EgadsPredictionTimeSeries;
+
+/**
+ * A class that parses and appends the period-over-period baseline queries to
+ * an Egads time series, then runs the prediction on the given data. 
+ * 
+ * TODO - re-use bits of this for other models.
+ * 
+ * @since 3.0
+ */
+public class OlympicScoringBaseline {
+  public static final String MODEL_TAG_VALUE = "OlympicScoring";
+  
+  private static final Logger LOG = LoggerFactory.getLogger(
+      OlympicScoringBaseline.class);
+  
+  protected final OlympicScoringNode node;
+  protected final TimeSeriesId id;
+  protected final com.yahoo.egads.data.TimeSeries baseline;
+  protected long last_ts;
+  
+  OlympicScoringBaseline(final OlympicScoringNode node, 
+                         final TimeSeriesId id) {
+    this.node = node;
+    this.id = id;
+    baseline = new com.yahoo.egads.data.TimeSeries();
+  }
+  
+  /**
+   * Assume that we get time series in order from oldest to newest.
+   * @param series
+   * @param result
+   */
+  void append(final TimeSeries series, final QueryResult result) {
+    TypeToken<? extends TimeSeriesDataType> type = null;
+    for (final TypeToken<? extends TimeSeriesDataType> t : series.types()) {
+      if (t == NumericType.TYPE ||
+          t == NumericArrayType.TYPE ||
+          t == NumericSummaryType.TYPE) {
+        type = t;
+        break;
+      }
+    }
+    if (type == null) {
+      LOG.warn("Time series: " + series.id() + " did not have a supported "
+          + "numeric iterator.");
+      return;
+    }
+    
+    final Optional<TypedTimeSeriesIterator<?>> optional = series.iterator(type);
+    if (!optional.isPresent()) {
+      LOG.warn("Time series: " + series.id() + " did not have a supported "
+          + "numeric iterator.");
+      return;
+    }
+    
+    if (type == NumericType.TYPE) {
+      processNumeric(optional.get());
+    } else if (type == NumericArrayType.TYPE) {
+      processNumericArray(optional.get(), result);
+    } else if (type == NumericSummaryType.TYPE) {
+      processSummary(optional.get());
+    }
+  }
+
+  TimeSeries predict(final Properties properties) {
+    if (baseline.size() < 2) {
+      LOG.warn("Not enough data points to predict: " + baseline.size());
+      return null;
+    }
+    
+    final com.yahoo.egads.data.TimeSeries prediction = 
+        new com.yahoo.egads.data.TimeSeries();
+    final double[] results = new double[(int) node.predictionIntervals()];
+    
+    // fill the prediction with nans at the proper timestamps
+    long ts = node.predictionStart();
+    for (int i = 0; i < results.length; i++) {
+      try {
+        prediction.append(ts, Float.NaN);
+      } catch (Exception e) {
+        throw new RuntimeException(e);
```

Comment:
Converting checked to unchecked makes me nervous in a language that pushes hard to handle checked. Where do you intend this to be handled?

Reply:
These should be handled by the caller. I really wish EGADs didn't have the checked exceptions and I may yank them out.